test/integration/notifications&auth/notifications-activation.test.tsx

```diff
@@ -0,0 +1,209 @@
+import { integrationTestRender } from '../../lib/render-helpers';
+import * as backgroundConnection from '../../../ui/store/background-connection';
+import {
+  act,
+  fireEvent,
+  waitFor,
+  screen,
+  within,
+} from '@testing-library/react';
+import { createMockImplementation } from '../helpers';
+import { getMockedNotificationsState } from './data/notification-state';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+} from '../../../shared/constants/metametrics';
+
+jest.mock('../../../ui/store/background-connection', () => ({
+  ...jest.requireActual('../../../ui/store/background-connection'),
+  submitRequestToBackground: jest.fn(),
+  callBackgroundMethod: jest.fn(),
+}));
+
+const backgroundConnectionMocked = {
+  onNotification: jest.fn(),
+};
+
+const mockedBackgroundConnection = jest.mocked(backgroundConnection);
+
+const setupSubmitRequestToBackgroundMocks = (
+  mockRequests?: Record<string, unknown>,
+) => {
+  mockedBackgroundConnection.submitRequestToBackground.mockImplementation(
+    createMockImplementation({
+      ...(mockRequests ?? {}),
+    }),
+  );
+};
+
+const trackNotificationsActivatedMetaMetricsEvent = async (
+  actionType: string,
+  profileSyncEnabled: boolean,
+) => {
+  await waitFor(() => {
+    const action =
+      mockedBackgroundConnection.submitRequestToBackground.mock.calls?.find(
+        (call) =>
+          call[0] === 'trackMetaMetricsEvent' &&
+          call[1]?.[0].category ===
+            MetaMetricsEventCategory.NotificationsActivationFlow &&
+          call[1]?.[0].properties.action_type === actionType,
+      );
+
+    expect(action?.[0]).toBe('trackMetaMetricsEvent');
+
+    const [event] = action?.[1] as unknown as [
+      {
+        event: string;
+        category: string;
+        properties: Record<string, unknown>;
+      },
+    ];
+
+    expect(event?.event).toBe(MetaMetricsEventName.NotificationsActivated);
+    expect(event?.category).toBe(
+      MetaMetricsEventCategory.NotificationsActivationFlow,
+    );
+    expect(event?.properties).toMatchObject({
+      action_type: actionType,
+      is_profile_syncing_enabled: profileSyncEnabled,
+    });
+  });
```

Comment:
Just a bit of exploration. I was seeing if we can clean up this assertion, I want your thoughts.
```ts
const trackNotificationsActivatedMetaMetricsEvent = async (
  actionType: string,
  profileSyncEnabled: boolean,
) => {
  const wasBackgroundEventFired =
    mockedBackgroundConnection.submitRequestToBackground.mock.calls?.some(
      (call) => {
        try {
          const [method, [event] = []] = call;
          expect(method).toBe('trackMetaMetricsEvent');
          expect(event).toStrictEqual(
            expect.objectContaining({
              event: MetaMetricsEventName.NotificationsActivated,
              category: MetaMetricsEventCategory.NotificationsActivationFlow,
              properties: {
                action_type: actionType,
                is_profile_syncing_enabled: profileSyncEnabled,
              },
            }),
          );
          return true;
        } catch {
          return false;
        }
      },
    );

  expect(wasBackgroundEventFired).toBe(true);
};
```

Reply:
Pros:
- I like the `const [method, [event] = []] = call;` destructor, keeps it nice and clean.
- I like the `expect().toStrictEqual`, I think it keeps it organised.

Cons:
- We lose the failure information if the assertion fails. E.g. the only thing we see in the console would be `expect(wasBackgroundEventFired).toBe(true);`, the internals won't be logged.
  - I think this is okay, as then devs can test this locally. But if this is a CI failure then it would make it hard to debug.